python/src/spark_rapids_ml/params.py

```diff
@@ -273,21 +303,18 @@ def copy(self: P, extra: Optional["ParamMap"] = None) -> P:
         instance: P = super().copy(extra)
         cuml_params = instance.cuml_params.copy()
 
+        instance._cuml_params = cuml_params
```

Comment:
For future reference maybe add a docstring for copy()?

Reply:
added